server/src/lsp-server.ts

```diff
@@ -123,6 +126,7 @@ export class LspServer {
 
         this.tspClient.start();
         this.tspClient.request(CommandTypes.Configure, {
+            hostInfo: 'coc-nvim',
```

Comment:
This isn't quite right as the coc-nvim behaviour includes a `path.resolve` call on a string prepended with `zipfile:` - I can't understand how this is useful to coc-nvim but it definitely doesn't work for us

https://github.com/yarnpkg/berry/blob/4b349329427aebc4460d8481ae8591a8e88ff4a2/.yarn/sdks/typescript/lib/tsserver.js#L68

Reply:
It's insane how yarn-specific this fix has to be. Even to the point of disguising as a specific editor...

I'm not saying it's unacceptable but it does feel gross indeed.
And if that part is kept, I hope it doesn't break other editors or at least makes things harder for those.

Reply:
Yeah, it does feel very yarn / vim specific passing through `zipfile:` prepended paths. We could make `hostInfo` configurable via CLI

Reply:
> This isn't quite right as the coc-nvim behaviour includes a `path.resolve` call on a string prepended with `zipfile:` - I can't understand how this is useful to coc-nvim but it definitely doesn't work for us
> 
> [yarnpkg/berry@`4b34932`/.yarn/sdks/typescript/lib/tsserver.js#L68](https://github.com/yarnpkg/berry/blob/4b349329427aebc4460d8481ae8591a8e88ff4a2/.yarn/sdks/typescript/lib/tsserver.js#L68)

Through `initializationOptions` would make most sense.

Reply:
Of course! Let me see how that looks